pkgs/development/python-modules/python-snap7/default.nix

```diff
@@ -21,13 +25,14 @@ buildPythonPackage rec {
     hash = "sha256-mcdzgR0z2P5inK9Q+ZQhP5H8vZSaPbRCSEnt+wzG+ro=";
   };
 
-  prePatch = ''
-    substituteInPlace snap7/common.py \
-      --replace "lib_location = None" "lib_location = '${snap7}/lib/libsnap7.so'"
-  '';
-
   build-system = [ setuptools ];
 
+  dependencies = [ click ];
+
+  postInstall = ''
+    ln -s ${snap7}/lib $out/${python.sitePackages}/snap7/lib
```

Comment:
The packages tries to load the bundled library from this location first.
(the PyPI wheel has a bundled .so in snap7/lib)

This is better, since we don't have to deal with .dylib and .so separately